frontend/src/pages/RunDetailsV2.tsx

```diff
@@ -214,6 +230,23 @@ export function RunDetailsV2(props: RunDetailsV2Props) {
 
           // TODO(zijianjoy): Wait backend to supply run parameters, so UI can show them.
         )}
+
+        {/* Pipeline Spec tab */}
+        {selectedTab === 2 && isTemplatePullSuccessFromRun && (
+          <div className={commonCss.codeEditor} data-testid={'spec-ir'}>
```

Comment:
`data-testid` is purely for testing, we should consider removing it when possible.

Reply:
Also, if existing code is using this approach, feel free to fix it wherever you see appropriate. Thank you!

Reply:
Thank you James for providing the following references.
1. Issue to react-ace: https://github.com/securingsincity/react-ace/issues/1289
2. Issue to React testing library: https://github.com/testing-library/react-testing-library/issues/793

Currently, the testing library doesn't support get `string value`. Thus, we can only use `test-id` in this case.